src/Diagnostics/VarianceDissipationComputations/VarianceDissipationComputations.jl

```diff
@@ -0,0 +1,126 @@
+module VarianceDissipationComputations
+ 
+export VarianceDissipation, get_dissipation_fields
+
+using Oceananigans.Grids: architecture
+using Oceananigans.Utils
+using Oceananigans.TimeSteppers
+using Oceananigans.Fields
+using Oceananigans.Fields: Field, VelocityFields
+using Oceananigans.Operators
+using Oceananigans.BoundaryConditions
+using Oceananigans.TurbulenceClosures: viscosity,
+                                       diffusivity, 
+                                       ScalarDiffusivity, 
+                                       ScalarBiharmonicDiffusivity,
+                                       AbstractTurbulenceClosure,
+                                       HorizontalFormulation,
+                                       _diffusive_flux_x,
+                                       _diffusive_flux_y,
+                                       _diffusive_flux_z
+
+using Oceananigans.Advection: _advective_tracer_flux_x, 
+                              _advective_tracer_flux_y, 
+                              _advective_tracer_flux_z
+
+using Oceananigans.Operators: volume
+using KernelAbstractions: @kernel, @index
+using Oceananigans: AbstractDiagnostic
+
+import Oceananigans: run_diagnostic!
+
+struct VarianceDissipation{P, K, A, D, S, G, T} <: AbstractDiagnostic
+    schedule :: T
+    advective_production :: P
+    diffusive_production :: K
+    advective_fluxes :: A
+    diffusive_fluxes :: D
+    previous_state :: S
+    gradient_squared :: G
+end
+
+include("dissipation_utils.jl")
+
+"""
+    VarianceDissipation(model; tracers=propertynames(model.tracers))
+
+Constructs a `VarianceDissipation` object for a given `model`. This function computes 
+the variance dissipation diagnostics for the specified tracers in the model.
+
+Argument
+=========
+
+- `model`: The model for which variance dissipation is to be computed. 
+  The model must use a `QuasiAdamsBashforth2TimeStepper` for its time-stepping scheme.
+
+Keyword Argument
+================
+
+- `tracers`: An optional argument specifying the tracers for which variance dissipation 
```

Comment:
How hard is it to generalize for momentum?

Reply:
For flux form momentum transport is rather trivial (the formulation is exaclty the same, we need to change only the `update_transport!` to cache interpolated advecting velocities.

For vector invariant momentum transport it is rather difficult, we are trying to figure out a way to have a numerical solution that closes the budgets, also in that case.